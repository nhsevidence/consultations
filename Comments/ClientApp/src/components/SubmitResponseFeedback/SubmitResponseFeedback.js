import React from "react";

export const SubmitResponseFeedback = (props) => {

	const {
		validToSubmit,
		unsavedIdsQty,
		organisationName,
		tobaccoDisclosure,
		respondingAsOrganisation,
		hasTobaccoLinks,
		showExpressionOfInterestSubmissionQuestion,
		organisationExpressionOfInterest,
		emailAddress,
		isOrganisationCommenter,
	} = props;

	let items = [];
	const emailRegex = /\S+@\S+\.\S+/;

	if (unsavedIdsQty)
		items.push("You have unsaved changes. Please save or delete before submitting your response");

	if (!validToSubmit)
		items.push("You have not saved any comments");

	if (isOrganisationCommenter) {
		if (emailAddress.length < 1) {
			items.push("You have not entered an email address");
		} else if (!emailRegex.test(emailAddress.toLowerCase())) {
			items.push("Email address is in an invalid format");
		}
	}

	if (!isOrganisationCommenter) {
		if (!respondingAsOrganisation)
			items.push("You have not stated whether you are submitting the response on behalf of an organisation");

		if (respondingAsOrganisation === "yes" && organisationName.length < 1)
			items.push("You have stated that you are responding on behalf of an organisation but you haven't entered the organisation name");

		if (respondingAsOrganisation === "yes" && showExpressionOfInterestSubmissionQuestion && organisationExpressionOfInterest === null)
			items.push("You have not disclosed whether your organisation would like to express an interest in formally supporting this quality standard");

		if (!hasTobaccoLinks)
			items.push("You have not disclosed whether you or the organisation you represent have links to the tobacco industry");

		if (hasTobaccoLinks === "yes" && tobaccoDisclosure.length < 1)
			items.push("You have indicated that you or the organisation you represent have links with the tobacco industry but you have not supplied any details");
	}

	if (!items.length) return null;

	return (
		<div data-qa-sel="Submit-response-feedback" className="SubmitResponseFeedback mb--d" id="SubmitResponseFeedback">
			<p className="SubmitResponseFeedback__title">You can't submit your response yet</p>
			<ul className="mt--0">
				{items.map(i => <li key={i}>{i}</li>)}
			</ul>
		</div>
	);

};
