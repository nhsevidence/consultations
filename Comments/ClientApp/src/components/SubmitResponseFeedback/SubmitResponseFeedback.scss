.SubmitResponseFeedback {
  padding: 0 rem($spacing-medium);
  border: 3px solid $colour-error;
  background-color: $colour-error-background;
  &__title {
    color: $colour-error;
    font-weight: bold;
  }
}
