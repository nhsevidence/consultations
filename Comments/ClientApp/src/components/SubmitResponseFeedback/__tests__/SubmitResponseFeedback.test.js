import { SubmitResponseFeedback } from "../SubmitResponseFeedback";

describe("[ClientApp", function () {
	describe("Submit response feedback alert box", () => {

		const fakeProps = {
			validToSubmit: false,
			unsavedIdsQty: 1,
			organisationName: "",
			tobaccoDisclosure: "",
			respondingAsOrganisation: false,
			hasTobaccoLinks: false,
		};

		it("should match the snapshot when almost everything that can be wrong, is", function () {
			let props = Object.assign({}, fakeProps);
			const wrapper = SubmitResponseFeedback(props);
			expect(wrapper).toMatchSnapshot();
		});

		it("should match the snapshot when all is good except for no saved comments", function () {
			let props = {
				validToSubmit: true,
				unsavedIdsQty: 0,
				organisationName: "",
				tobaccoDisclosure: "",
				respondingAsOrganisation: "no",
				hasTobaccoLinks: "no",
			};
			const wrapper = SubmitResponseFeedback(props);
			expect(wrapper).toMatchSnapshot();
		});

		it("should match the snapshot when comments are saved", function () {
			let props = Object.assign({}, fakeProps);
			props.validToSubmit = true;
			const wrapper = SubmitResponseFeedback(props);
			expect(wrapper).toMatchSnapshot();
		});

		it("should match the snapshot when responding as organisation is yes but there is no name", function () {
			let props = Object.assign({}, fakeProps);
			props.respondingAsOrganisation = "yes";
			const wrapper = SubmitResponseFeedback(props);
			expect(wrapper).toMatchSnapshot();
		});

		it("should match the snapshot when tobacco disclosure is yes but there is no further detail", function () {
			let props = Object.assign({}, fakeProps);
			props.hasTobaccoLinks = "yes";
			const wrapper = SubmitResponseFeedback(props);
			expect(wrapper).toMatchSnapshot();
		});

		it("should match the snapshot when everything is good apart from 2 unsaved changes", function () {
			let props = {
				validToSubmit: true,
				unsavedIdsQty: 2,
				organisationName: "",
				tobaccoDisclosure: "",
				respondingAsOrganisation: true,
				hasTobaccoLinks: true,
			};
			const wrapper = SubmitResponseFeedback(props);
			expect(wrapper).toMatchSnapshot();
		});

		it("should match the snapshot when everything is good apart from 2 unsaved changes", function () {
			let props = {
				validToSubmit: true,
				unsavedIdsQty: 2,
				organisationName: "",
				tobaccoDisclosure: "",
				respondingAsOrganisation: true,
				hasTobaccoLinks: true,
			};
			const wrapper = SubmitResponseFeedback(props);
			expect(wrapper).toMatchSnapshot();
		});

		it("should return null if everything is fine", function () {
			let props = {
				validToSubmit: true,
				unsavedIdsQty: 0,
				organisationName: "Org",
				tobaccoDisclosure: "Heavy smoker",
				respondingAsOrganisation: "yes",
				hasTobaccoLinks: "yes",
			};
			const wrapper = SubmitResponseFeedback(props);
			expect(wrapper).toEqual(null);
		});

	});
});
