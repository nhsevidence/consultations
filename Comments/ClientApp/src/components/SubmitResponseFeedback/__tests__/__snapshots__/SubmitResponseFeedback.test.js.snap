// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`[ClientApp Submit response feedback alert box should match the snapshot when all is good except for no saved comments 1`] = `null`;

exports[`[ClientApp Submit response feedback alert box should match the snapshot when almost everything that can be wrong, is 1`] = `
<div
  className="SubmitResponseFeedback mb--d"
  data-qa-sel="Submit-response-feedback"
  id="SubmitResponseFeedback"
>
  <p
    className="SubmitResponseFeedback__title"
  >
    You can't submit your response yet
  </p>
  <ul
    className="mt--0"
  >
    <li>
      You have too many answers for the question "Which areas will have the biggest impact on practice and be challenging to implement? Please say for whom and why."
    </li>
    <li>
      You have unsaved changes. Please save or delete before submitting your response
    </li>
    <li>
      You have not saved any comments
    </li>
    <li>
      You have not stated whether you are submitting the response on behalf of an organisation
    </li>
    <li>
      You have not disclosed whether you or the organisation you represent have links to the tobacco industry
    </li>
  </ul>
</div>
`;

exports[`[ClientApp Submit response feedback alert box should match the snapshot when comments are saved 1`] = `
<div
  className="SubmitResponseFeedback mb--d"
  data-qa-sel="Submit-response-feedback"
  id="SubmitResponseFeedback"
>
  <p
    className="SubmitResponseFeedback__title"
  >
    You can't submit your response yet
  </p>
  <ul
    className="mt--0"
  >
    <li>
      You have too many answers for the question "Which areas will have the biggest impact on practice and be challenging to implement? Please say for whom and why."
    </li>
    <li>
      You have unsaved changes. Please save or delete before submitting your response
    </li>
    <li>
      You have not stated whether you are submitting the response on behalf of an organisation
    </li>
    <li>
      You have not disclosed whether you or the organisation you represent have links to the tobacco industry
    </li>
  </ul>
</div>
`;

exports[`[ClientApp Submit response feedback alert box should match the snapshot when everything is good apart from 2 unsaved changes 1`] = `
<div
  className="SubmitResponseFeedback mb--d"
  data-qa-sel="Submit-response-feedback"
  id="SubmitResponseFeedback"
>
  <p
    className="SubmitResponseFeedback__title"
  >
    You can't submit your response yet
  </p>
  <ul
    className="mt--0"
  >
    <li>
      You have unsaved changes. Please save or delete before submitting your response
    </li>
  </ul>
</div>
`;

exports[`[ClientApp Submit response feedback alert box should match the snapshot when everything is good apart from 2 unsaved changes 2`] = `
<div
  className="SubmitResponseFeedback mb--d"
  data-qa-sel="Submit-response-feedback"
  id="SubmitResponseFeedback"
>
  <p
    className="SubmitResponseFeedback__title"
  >
    You can't submit your response yet
  </p>
  <ul
    className="mt--0"
  >
    <li>
      You have unsaved changes. Please save or delete before submitting your response
    </li>
  </ul>
</div>
`;

exports[`[ClientApp Submit response feedback alert box should match the snapshot when everything is good apart from too many answers to one question 1`] = `
<div
  className="SubmitResponseFeedback mb--d"
  data-qa-sel="Submit-response-feedback"
  id="SubmitResponseFeedback"
>
  <p
    className="SubmitResponseFeedback__title"
  >
    You can't submit your response yet
  </p>
  <ul
    className="mt--0"
  >
    <li>
      You have too many answers for the question "Which areas will have the biggest impact on practice and be challenging to implement? Please say for whom and why."
    </li>
  </ul>
</div>
`;

exports[`[ClientApp Submit response feedback alert box should match the snapshot when responding as organisation is yes but there is no name 1`] = `
<div
  className="SubmitResponseFeedback mb--d"
  data-qa-sel="Submit-response-feedback"
  id="SubmitResponseFeedback"
>
  <p
    className="SubmitResponseFeedback__title"
  >
    You can't submit your response yet
  </p>
  <ul
    className="mt--0"
  >
    <li>
      You have too many answers for the question "Which areas will have the biggest impact on practice and be challenging to implement? Please say for whom and why."
    </li>
    <li>
      You have unsaved changes. Please save or delete before submitting your response
    </li>
    <li>
      You have not saved any comments
    </li>
    <li>
      You have stated that you are responding on behalf of an organisation but you haven't entered the organisation name
    </li>
    <li>
      You have not disclosed whether you or the organisation you represent have links to the tobacco industry
    </li>
  </ul>
</div>
`;

exports[`[ClientApp Submit response feedback alert box should match the snapshot when tobacco disclosure is yes but there is no further detail 1`] = `
<div
  className="SubmitResponseFeedback mb--d"
  data-qa-sel="Submit-response-feedback"
  id="SubmitResponseFeedback"
>
  <p
    className="SubmitResponseFeedback__title"
  >
    You can't submit your response yet
  </p>
  <ul
    className="mt--0"
  >
    <li>
      You have too many answers for the question "Which areas will have the biggest impact on practice and be challenging to implement? Please say for whom and why."
    </li>
    <li>
      You have unsaved changes. Please save or delete before submitting your response
    </li>
    <li>
      You have not saved any comments
    </li>
    <li>
      You have not stated whether you are submitting the response on behalf of an organisation
    </li>
    <li>
      You have indicated that you or the organisation you represent have links with the tobacco industry but you have not supplied any details
    </li>
  </ul>
</div>
`;
